packages/toolpad-components/src/TextField.tsx

```diff
@@ -5,24 +5,29 @@ import {
   BoxProps,
 } from '@mui/material';
 import { createComponent } from '@mui/toolpad-core';
+import { useField } from 'formik';
 import { SX_PROP_HELPER_TEXT } from './constants';
 
 export type TextFieldProps = Omit<MuiTextFieldProps, 'value' | 'onChange'> & {
   value: string;
   onChange: (newValue: string) => void;
   alignItems?: BoxProps['alignItems'];
   justifyContent?: BoxProps['justifyContent'];
+  name: string;
 };
 
-function TextField({ defaultValue, onChange, ...props }: TextFieldProps) {
+function TextField({ defaultValue, onChange, value, ...props }: TextFieldProps) {
+  const [field] = useField(props.name);
```

Comment:
So every `TextField` is using form context?
Wouldn't it be better to pass `field.onChange` to the `onChange` prop, and `field.value` to the `value` prop of this component somehow? (probably in the runtime)
Same for the other components where you followed this approach.

Reply:
I'm not sure about that, I guess one way to do that could be adding a "global form content" where we could catch all values and still keep using them for individual components, then as soon as we add form it could capture values inside it's own context.

However the main reason why I didn't do something like that is that I wanted to limited the surface of the changes and see whether it's worth to continue building on top. So I'd say it's up for a discussion

Reply:
Yeah, it wouldn't be an easy change.
It can be up to you, I don't mind us doing things in steps as long as it won't cause any extra trouble.